app/src/main/java/com/example/a17255/a8project/GamePlay.java

```diff
@@ -132,7 +347,52 @@ protected void onCreate(Bundle savedInstanceState) {
 //        setContentView(R.layout.activity_game_play);
         setContentView(myCanvas);
 
+
+        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+        accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+
         gestureDetectorCompat = new GestureDetectorCompat(this, this);
     }
 
+
+    public void confirmFireMissiles() {
+        DialogFragment newFragment = new WinDialogFragment(GamePlay.this, score);
+        newFragment.show(getFragmentManager(), "win");
+    }
+
+    public void saveScoreInSharedPreferences(int score) {
+        SharedPreferences sharedPref = GamePlay.this.getSharedPreferences("abc",Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPref.edit();
+
+
+
+        int temp = sharedPref.getInt("index", 0);
+
+        editor.putInt(Integer.toString(temp), score);
+        editor.commit();
+
+        temp++;
+        editor.putInt("index", temp);
+        editor.commit();
```

Comment:
no need to commit 2 times
Also it is almost always better to use apply instead of commit